src/probnum/linalg/solvers/__init__.py

```diff
@@ -0,0 +1,31 @@
+"""Probabilistic linear solvers.
+
+Compositional implementation of probabilistic linear solvers. The
+classes and methods in this subpackage allow the creation of custom
+iterative methods for the solution of linear systems.
+"""
+
+from probnum.linalg.solvers.matrixbased import (
+    AsymmetricMatrixBasedSolver,
+    MatrixBasedSolver,
+    ProbabilisticLinearSolver,
+    SymmetricMatrixBasedSolver,
+)
+from probnum.linalg.solvers.solutionbased import SolutionBasedSolver
+
+from . import beliefs, policies
+from ._state import LinearSolverState
```

Comment:
While I think from a developer perspective this is a nice idea, i decided against it. The reason was that an inner class doesn't seem to play nice with `automodapi` for the documentation. I couldn't find a way to actually have it show up in the documentation with our current setup. Creating a custom solution just to have `ProbabilisticLinearSolver.State` instead of `ProbabilisticLinearSolverState` is nor worth the time imo.